fs/backgroundfetcher/background_fetcher.go

```diff
@@ -168,3 +179,18 @@ func (bf *BackgroundFetcher) Run(ctx context.Context) error {
 		}
 	}
 }
+
+func (bf *BackgroundFetcher) emitWorkQueueMetric(ctx context.Context) {
+	for {
+		select {
+		case <-bf.closeChan:
+			return
+		case <-ctx.Done():
+			return
+		default:
+			// background fetcher is at the snapshotter's fs level, so no image digest as key
+			commonmetrics.AddImageOperationCount(commonmetrics.BackgroundFetchWorkQueueSize, "", int32(len(bf.workQueue)))
+			time.Sleep(bf.emitMetricPeriod)
```

Comment:
nit: go has `time.Ticker` for this sort of select block so the sleep doesn't prevent you from receiving on other channels:

```
ticker := time.NewTicker(b.emitMetricPeriod)
select {
    // other cases
    case <- ticker.C:
    // emit metric
}
```